lib/compat/wordpress-6.5/blocks.php

```diff
@@ -67,6 +67,70 @@ function gutenberg_block_bindings_replace_html( $block_content, $block_name, str
 		switch ( $block_type->attributes[ $attribute_name ]['source'] ) {
 			case 'html':
 			case 'rich-text':
+				if ( 'core/image' === $block_name && 'caption' === $attribute_name ) {
```

Comment:
I think you will have to move it to `compat/wordpress-6.6/` because that functionality is also missing in WordPress 6.5. If the code stays only here, then after support for WordPress 6.4 gets dropped the polyfill for WP 6.5 will be missing.